opal/corelib/io.rb

```diff
@@ -3,6 +3,25 @@ class IO
   SEEK_CUR = 1
   SEEK_END = 2
 
+  attr_reader :eof
+  attr_reader :lineno
+
+  def initialize
+    @eof = false
+    @lineno = 0
+  end
+
+  def read
+    if @eof
+      ''
+    else
+      res = File.read(@path)
```

Comment:
I don't think this should be coupled to File, looking at Ruby docs it says it mimics `fread()` for which the input can be any file-like unix device, maybe we should rely on something pluggable like it's done with `write_proc`. 

`read_proc` maybe?

https://github.com/Mogztter/opal/blob/841af9d1a1f43eb3f0817dfa5ff14169ca5a0d3d/opal/corelib/io.rb#L33-L38
https://github.com/Mogztter/opal/blob/841af9d1a1f43eb3f0817dfa5ff14169ca5a0d3d/opal/corelib/io.rb#L91-L92